component/core/src/main/java/org/exoplatform/social/core/metadata/MetadataServiceImpl.java

```diff
@@ -223,7 +223,13 @@ public MetadataItem deleteMetadataItem(long itemId, long userIdentityId) throws
 
   @Override
   public void deleteMetadataItemsByObject(MetadataObject object) {
+    List<MetadataItem> metadataItems = this.metadataStorage.getMetadataItemsByObject(object);
     this.metadataStorage.deleteMetadataItemsByObject(object);
+    try {
+      this.listenerService.broadcast("social.metadataItem.deleted", 0l, metadataItems);
+    } catch (Exception e) {
+      LOG.warn("Error while broadcasting event for metadata delete", e);
```

Comment:
Why not using `broadcastDeleted` for each item ? In fact this will ensure to not have bugs in other addons that assumes that this event uses an object of Type `MetadataItem` and not `List`